src/solidity-frontend/solidity_convert.cpp

```diff
@@ -0,0 +1,1831 @@
+      found_contract_def = true;
+      // pattern-based verification
+      assert(itr->contains("nodes"));
+      auto pattern_check = std::make_unique<pattern_checker>((*itr)["nodes"], sol_func, msg);
+      if(pattern_check->do_pattern_check())
+        return true; // 'true' indicates something goes wrong.
+    }
+  }
+  assert(found_contract_def);
+
+  // reasoning-based verification
+  //assert(!"Continue with symbol annotations");
+  index = 0;
+  itr = nodes.begin();
+  for (; itr != nodes.end(); ++itr, ++index)
+  {
+    std::string node_type = (*itr)["nodeType"].get<std::string>();
+    if (node_type == "ContractDefinition") // rule source-unit
+    {
+      if(convert_ast_nodes(*itr))
+        return true; // 'true' indicates something goes wrong.
+    }
+  }
+
+  return false; // 'false' indicates successful completion.
+}
+
+bool solidity_convertert::convert_ast_nodes(const nlohmann::json &contract_def)
+{
+  unsigned index = 0;
+  nlohmann::json ast_nodes = contract_def["nodes"];
+  nlohmann::json::iterator itr = ast_nodes.begin();
+  for (; itr != ast_nodes.end(); ++itr, ++index)
+  {
+    nlohmann::json ast_node = *itr;
+    std::string node_name = ast_node["name"].get<std::string>();
+    std::string node_type = ast_node["nodeType"].get<std::string>();
+    printf("@@ Converting node[%u]: name=%s, nodeType=%s ...\n",
+        index, node_name.c_str(), node_type.c_str());
+    //print_json_array_element(ast_node, node_type, index);
+    exprt dummy_decl;
+    if(get_decl(ast_node, dummy_decl))
+      return true;
+  }
+
+  assert(current_functionDecl == nullptr);
+
+  return false;
+}
+
+bool solidity_convertert::get_decl(const nlohmann::json &ast_node, exprt &new_expr)
+{
+  new_expr = code_skipt();
+
+  if (!ast_node.contains("nodeType"))
+    assert(!"Missing \'nodeType\' filed in ast_node");
+
+  SolidityGrammar::ContractBodyElementT type = SolidityGrammar::get_contract_body_element_t(ast_node);
+
+  // based on each element as in Solidty grammar "rule contract-body-element"
+  switch(type)
+  {
+    case SolidityGrammar::ContractBodyElementT::StateVarDecl:
+    {
+      return get_var_decl(ast_node, new_expr); // rule state-variable-declaration
+    }
+    case SolidityGrammar::ContractBodyElementT::FunctionDef:
+    {
+      return get_function_definition(ast_node, new_expr); // rule function-definition
+    }
+    default:
+    {
+      assert(!"Unimplemented type in rule contract-body-element");
+      return true;
+    }
+  }
+
+  return false;
+}
+
+bool solidity_convertert::get_var_decl_stmt(const nlohmann::json &ast_node, exprt &new_expr)
+{
+  // For rule variable-declaration-statement
+  new_expr = code_skipt();
+
+  if (!ast_node.contains("nodeType"))
+    assert(!"Missing \'nodeType\' filed in ast_node");
+
+  SolidityGrammar::VarDeclStmtT type = SolidityGrammar::get_var_decl_stmt_t(ast_node);
+  printf("	@@@ got Variable-declaration-statement: SolidityGrammar::VarDeclStmtT::%s\n",
```

Comment:
Why don't we use our messaging interface based on the fmt library?

Reply:
ACK. I'll replace them with `msg.debug(fmt::format(.))`.